src/universal/decorators/requireAuthAndRole/requireAuthAndRole.js

```diff
@@ -0,0 +1,43 @@
+import React, {Component, PropTypes} from 'react';
+import {push} from 'react-router-redux';
+import getAuth from 'universal/redux/getAuth';
+import {cashay} from 'cashay';
+import jwtDecode from 'jwt-decode';
+import {error as showError} from 'universal/modules/notifications/ducks/notifications';
+
+const unauthorized = {
+  title: 'Unauthorized',
+  message: 'Hey! You\'re not supposed to be there. Bringing you someplace safe.',
+};
+
+export default role => {
+  return (ComposedComponent) => {
+    return class RequireAuthAndRole extends Component {
+      static propTypes = {
+        lastPathName: PropTypes.string,
+      }
+
+      render() {
+        // no need to check for expired tokens here, right?
+        // We check initial validation in the client.js. Our job should be to keep them logged in.
+        const auth = getAuth();
+        const authObj = auth.authToken && jwtDecode(auth.authToken);
+        if (authObj) {
+          if (role && authObj.rol === role) {
+            // We had a role to check, and role checks out:
+            return <ComposedComponent {...this.props} auth={auth} />;
+          } else if (role === undefined) {
+            // We were looking for any authenticated user only:
+            return <ComposedComponent {...this.props} auth={auth} />;
+          }
+        }
+        // TODO: redirect back from whence they came
```

Comment:
This is as easy as `dispatch(goBack())`, but I think you got it right by going to root.

Reason being, if they somehow got to a place where only a parabol employee should be, they didn't get there via a link (cause that's on us if we show them a link!). They got there by directly pasting in a url.

All of this to say, if we tell them to `goBack`, then they won't get a notification, because the `goBack` will navigate them away from our site. But if we send them to root, we can show them a notification. Will fix in upcoming PR.